src/components/VideoProvider/useBackgroundSettings/useBackgroundSettings.ts

```diff
@@ -65,13 +67,68 @@ export const backgroundConfig = {
   images,
 };
 
-// TODO : Add video processing logic after backgroundSettings change
-// useEffect hooks, etc ...
+const bgLibSettings = {
+  assetsPath: '/virtualbackground',
+};
+
+const BG_SETTINGS_KEY = 'bgsettings5';
+
+let blurProcessor: GaussianBlurBackgroundProcessor;
+
+export default function useBackgroundSettings(room: Room | undefined | null) {
+  let currentSettings: BackgroundSettings;
+  const localStorageSetting = window.localStorage.getItem(BG_SETTINGS_KEY);
+  currentSettings = localStorageSetting ? JSON.parse(localStorageSetting!) : { type: 'none', index: 0 };
+
+  const [backgroundSettings, setBackgroundSettings] = useState<BackgroundSettings>(currentSettings);
+
+  const updateSettings = (settings: BackgroundSettings): void => {
+    setBackgroundSettings(settings);
+    window.localStorage.setItem(BG_SETTINGS_KEY, JSON.stringify(settings));
+  };
+
+  const updateBackgroundSettings = useCallback(
+    async (settings: BackgroundSettings, reset?: boolean, force?: boolean, newRoom?: Room) => {
+      // reset, used for resetting background settings when camera is off/no video tracks
+      let currentRoom = room;
+      if (reset) {
+        updateSettings({ type: 'none' });
+        return;
+      }
+      currentRoom = currentRoom || newRoom;
+      if (!currentRoom) {
+        return;
+      }
+
+      const videoTrackValue = Array.from(currentRoom.localParticipant.videoTracks.values())[0];
+      if (!videoTrackValue) {
+        return;
+      }
+      const videoTrack = videoTrackValue.track;
+      const { type } = settings;
+      if (type !== backgroundSettings.type || force) {
```

Comment:
Why do we need `force` again?

Reply:
There's also a lot of duplicated checks in this if block. I understand you are using my prototype as a reference 😅 . I think this block can be simplified.

Move the initialization outside of updateBackgroundSettings. You might need to wrap it in an useEffect.
```ts
if (!blurProcessor) {
  blurProcessor = new GaussianBlurBackgroundProcessor({ ...bgLibSettings });
  await blurProcessor.loadModel();
}
```

Then you can shrink your if block to something like this
```ts
if (type !== backgroundSettings.type || force) {
  if (videoTrack.processor) {
    videoTrack.removeProcessor(videoTrack.processor);
  }
  if (type === 'blur') {
    videoTrack.addProcessor(blurProcessor);
  }
}
```